### PR TITLE
Data stream refactor

### PR DESCRIPTION
**Pull request by annastopel**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:
In this refactor all dataStream selection logic is taken out of dataStream.go file and placed in dedicated file with respect to endpoint type. 
**Release note**:

```release-note
None
```

### CODE REVIEW

**Comment by annastopel** [comment]
> This appears to be based on the registry import PR branch. Should be based on master. Maybe hold off on this until registry import is merged.

@mhenriks  now it is based on master, reviews are welcome

File: pkg/importer/registryDataStreamer.go [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+func (i *registryData) cleanup() error {
+	glog.V(1).Infof("registryData - deleting all the data")
+	if i.file != nil {
+		i.file.Close()
+		i.file = nil
+	}
+	if len(i.dataDir) > 0 {
+		err := os.RemoveAll(i.dataDir)
+		if err != nil {
+			glog.Errorf(" Failed removing directory ", err.Error())
```
Comment (j-griffith) [inline pkg/importer/registryDataStreamer.go]: You should add the formatting directive here to Errorf

File: pkg/importer/s3DataStreamer.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+)
+
+//s3DataStreamer
+type s3DataStreamer struct {
+	secKey    string
+	accessKey string
+	url       *url.URL
+}
+
+//S3DataStreamer - -creates tsreamer that fetches file from S3 source
```
Comment (igoihman) [inline pkg/importer/s3DataStreamer.go]: creates streamer

File: pkg/importer/dataFormat.go [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+type DataFormatInterface interface {
+	setArchived(archived bool)
+	setDirectStreaming(encrypted bool, remoteStreaming bool, url *url.URL)
+	setLocalPath(localPath string)
+	store(reader io.ReadCloser, destPath string, scratchSpace string) error
+	resizeImage(dest, imageSize string, totalTargetSpace int64) error
+}
+
+// GenericDataFormat - generic data format
+type GenericDataFormat struct {
```
Comment (igoihman) [inline pkg/importer/dataFormat.go]: Maybe `DataFormat` or `BaseDataFormat`?

**Comment by awels** [comment]
I think this is super seeded by PR #721 please close if you agree.

**Comment by annastopel** [comment]
Agreed

**Comment by awels** [comment]
Closing per agreement.

**Pull request closed without merging**